app/stac_api/models.py

```diff
@@ -197,27 +190,469 @@ def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asse
                 "Error when updating collection's summaries values due to asset update."
             ))
 
-    def update_temporal_extent(self, item_properties_start_datetime, item_properties_end_datetime):
+    def insert_item(
+        self, old_start_datetime, new_start_datetime, old_end_datetime, new_end_datetime, item_id
+    ):
         '''
-        updates the collection's temporal extent when item's are update.
-        :param item_properties_datetime: item's value for properties_datetime.
-        This function checks, if the corresponding parameter of the collection
-        needs to be updated. If so, it will be either updated or an error will
-        be raised, if updating fails.
+        This function is called from within update_temporal_extent() when a new
+        item is inserted to the collection.
+        :param old_start_datetime: item's old value for properties_start_datetime
+        (in this case None, as it is a new item)
+        :param new_start_datetime: item's updated value for properties_start_datetime
+        :param old_end_datetime: item's old value for properties_end_datetime
+        (in this case None, as it is a new item)
+        :param new_end_datetime: item's updated value for properties_end_datetime
+        :param item_id: the id of the item being treated
         '''
-
+        logger.debug('Inserting item %s in collection %s', item_id, self.collection_name)
         if self.cache_start_datetime is None:
-            self.cache_start_datetime = item_properties_start_datetime
-            self.save()
-        elif item_properties_start_datetime < self.cache_start_datetime:
-            self.cache_start_datetime = item_properties_start_datetime
-            self.save()
-        elif self.cache_end_datetime is None:
-            self.cache_end_datetime = item_properties_end_datetime
-            self.save()
-        elif item_properties_end_datetime > self.cache_end_datetime:
-            self.cache_end_datetime = item_properties_end_datetime
-            self.save()
+            # first item in collection, as cache_start_datetime is None:
+            self.cache_start_datetime = new_start_datetime
+        elif self.cache_start_datetime > new_start_datetime:
+            # new item starts earlier that current collection range starts
+            self.cache_start_datetime = new_start_datetime
+
+        if self.cache_end_datetime is None:
+            #first item in collection, as cache_start_datetime is None
+            self.cache_end_datetime = new_end_datetime
+        elif self.cache_end_datetime < new_end_datetime:
+            # new item starts after current collection's range ends
+            self.cache_end_datetime = new_end_datetime
+
+        self.save()
+
+    def update_item_start_datetime(
```

Comment:
method name is confusing; `update_extent_item_update_start_datetime`

Reply:
renamed the method's here and in all other places